### ISSUE TITLE
Right-Click on blocks causes client to crash

### ISSUE THREAD

**Issue by qtGenn** [MehVahdJukaar/FarmersDelightRefabricated#105]
### Minecraft version

1.21.1

### Farmer's Delight version

~~2.1.1~~ 2.2.1

### Fabric Loader version

Quilt 0.27.0-beta.1

### Fabric API version

0.107.0

### Description

Right-clicking non-interactive blocks causes client to crash while causing a "[Netty Server IO #9/ERROR]: Exception occurred in netty pipeline" error on the server's console.

### Steps to reproduce

1. Right-click non-interactive block (stone, wood, glass, dirt...)
2. Crash

### Mod list

Client:
  [✔] actuallyharvest-1.21.1-1.1.1.1-FABRIC
  [✔] alloy-forgery-2.4.1+1.21
  [✔] AmbientSounds_FABRIC_v6.1.3_mc1.21.1
  [✔] animatica-0.6.1+1.21
  [✔] appleskin-fabric-mc1.21-3.0.6
  [✔] archers-2.1.1+1.21.1
  [✔] attributefix-fabric-1.21.1-21.1.2
  [✔] azurelibarmor-fabric-1.21.1-2.3.13
  [✔] BadOptimizations-2.2.0-1.21.1
  [✔] betterarcheology-1.2.0-1.21
  [✔] betterdays-1.21.1-3.1.1.1-FABRIC
  [✔] BetterF3-11.0.3-Fabric-1.21.1
  [✔] BetterGrassify-1.3.0+fabric.1.21.1
  [✔] BetterPingDisplay-Fabric-1.21.1-1.1.1
  [✔] bookshelf-fabric-1.21.1-21.1.13
  [✔] c2me-fabric-mc1.21.1-0.3.0+alpha.0.295
  [✔] cardinal-components-api-6.1.1
  [✔] citresewn-1.2.2+1.21
  [✔] cloth-config-15.0.140-fabric
  [✔] CollisionFix-1.0.1
  [✔] comforts-fabric-9.0.3+1.21.1
  [✔] connectedglass-1.1.12-fabric-mc1.21
  [✔] CreativeCore_FABRIC_v2.12.22_mc1.21.1
  [✔] Creeperoverhaul-fabric-1.21-4.0.5
  [✔] Debugify-1.21.1+1.0
  [✘] dehydration-1.3.7.jar (disabled)
  [✔] DisableAccessibilityScreen-1.0.1
  [✔] emi-1.1.18+1.21.1+fabric
  [✔] entity_model_features_fabric_1.21-2.2.6
  [✔] entity_texture_features_fabric_1.21.1-6.2.7
  [✔] fabric-api-0.107.0+1.21.1
  [✔] fabric-language-kotlin-1.12.3+kotlin.2.0.21
  [✔] fabricskyboxes-0.7.4+mc1.21
  [✔] FarmersDelight-1.21.1-2.2.1+refabricated
  [✔] fasterrandom-5.1.0
  [✔] ferritecore-7.0.2-hotfix-fabric
  [✔] fleshz-1.6.1
  [✔] ForgeConfigAPIPort-v21.1.1-1.21.1-Fabric
  [✔] frostiful-1.3.3
  [✔] fusion-1.1.1-fabric-mc1.21
  [✔] fzzy_config-0.5.6+1.21
  [✔] geckolib-fabric-1.21.1-4.7
  [✔] GlitchCore-fabric-1.21.1-2.1.0.0
  [✔] GpuTape-1.18x-1.21x-1.0.3
  [✔] handcrafted-fabric-1.21.1-4.0.2
  [✔] held-item-info-1.7.0
  [✔] ImmediatelyFast-Fabric-1.3.1+1.21.1
  [✔] ImmersiveThunder-1.21.1+1.2.2
  [✔] iris-fabric-1.8.0-beta.8+mc1.21.1
  [✔] Jade-1.21.1-Fabric-15.8.3
  [✔] jewelry-2.0.2+1.21.1
  [✔] journeymap-fabric-1.21.1-6.0.0-beta.29
  [✔] keybind_fix-1.0.1
  [✔] Kiwi-1.21-Fabric-15.1.2
  [✔] krypton-0.2.8
  [✔] Ksyxis-1.3.2
  [✔] lightweight-inventory-sorting-1.1.2+1.21
  [✔] lithium-fabric-mc1.21.1-0.13.1
  [✔] MapFrontiers-1.21-2.6.0p2-fabric
  [✔] modelfix-1.21-1.6-fabric
  [✔] modernfix-fabric-5.19.5+mc1.21.1
  [✔] ModernUI-Fabric-1.21.1-3.11.0.4-universal
  [✔] modmenu-11.0.3
  [✔] moremobvariants-fabric+1.21-1.3.1.1
  [✔] MRU-1.0.4+1.21+fabric
  [✔] mythicmetals-0.20.2+1.21
  [✔] NoChatReports-FABRIC-1.21.1-v2.9.1
  [✔] noxesium-2.3.3
  [✔] OverflowingBars-v21.1.1-1.21.1-Fabric
  [✔] owo-lib-0.12.15+1.21
  [✔] packetfixer-fabric-1.4.2-1.21-to-1.21.3
  [✔] paladins-2.1.1+1.21.1
  [✔] Palladium-1.21x-1.0.9.5
  [✔] particle_core-0.2.5+1.21
  [✔] particular-1.1.1+1.21
  [✔] placeholder-api-2.4.1+1.21
  [✔] player-animation-lib-fabric-2.0.0-alpha1+1.21
  [✔] polymorph-fabric-1.0.6+1.21.1
  [✔] polytone-1.21-2.5.5-fabric
  [✔] potatoptimize-0.0.1-dev.7
  [✔] PresenceFootsteps-1.10.0+1.21
  [✔] prickle-fabric-1.21.1-21.1.6
  [✔] purpurclient-1.21-37
  [✔] PuzzlesLib-v21.1.22-1.21.1-Fabric
  [✔] randommobsizes-fabric-1.21.1-2.2.2
  [✔] ranged_weapon_api-2.0.4+1.21.1
  [✔] rechiseled-1.1.6b-fabric-mc1.21
  [✔] reeses-sodium-options-fabric-1.8.0-beta.4+mc1.21.1
  [✔] resourcefulconfig-fabric-1.21-3.0.8
  [✔] resourcefullib-fabric-1.21-3.0.11
  [✔] rogues-2.1.0+1.21.1
  [✔] runes-1.0.3+1.21.1
  [✔] satin-2.0.0
  [✔] ScalableLux-0.1.0+fabric.26c6e72-all
  [✔] scorchful-0.12.1
  [✔] seasonhud-fabric-1.21.1-1.11.3
  [✔] SereneSeasons-fabric-1.21.1-10.1.0.1
  [✔] servercore-fabric-1.5.5+1.21.1
  [✔] ShieldAPI-2.0.2+1.21.1
  [✔] skinlayers3d-fabric-1.7.1-mc1.21
  [✔] SnowRealMagic-1.21.1-Fabric-11.0.12
  [✔] sodium-extra-fabric-0.6.0-beta.3+mc1.21.1
  [✔] sodiumextras-fabric-1.0.6-1.21.1
  [✔] sodium-fabric-0.6.0-beta.4+mc1.21.1
  [✔] sodiumleafculling-fabric-1.0.0-1.21.1
  [✔] sodiumoptionsapi-fabric-1.0.5-1.21.1
  [✔] sound-physics-remastered-fabric-1.21.1-1.4.6
  [✔] Sounds-2.2.0+1.21+fabric
  [✔] spell_engine-1.1.2+1.21.1
  [✔] spell_power-1.0.5+1.21.1
  [✔] supermartijn642configlib-1.1.8-fabric-mc1.21
  [✔] supermartijn642corelib-1.1.17c-fabric-mc1.21
  [✔] tatimod-0.1.1-1.21.1
  [✔] thermoo-4.2.3
  [✔] thermoo-patches-3.3
  [✔] threadtweak-fabric-0.1.5+mc1.21.1
  [✔] threatengl-quilt-1.3.3-release.1
  [✔] tide-fabric-1.21-1.3.4
  [✔] ToadLib-1.21x-1.1.6
  [✔] toms_storage_fabric-1.21-2.0.11
  [✔] travelersbackpack-fabric-1.21.1-10.1.3
  [✔] trinkets-3.10.0
  [✔] trowel-1.4.0
  [✔] villagerfix-1.0.4
  [✔] voicechat-fabric-1.21.1-2.5.25
  [✔] wizards-2.1.0+1.21.1
  [✔] YetAnotherConfigLib-3.6.1+1.21-fabric

Server:
[✔] actuallyharvest-1.21.1-1.1.1.1-FABRIC
  [✔] alloy-forgery-2.4.1+1.21
  [✔] alternate-current-mc1.21-1.9.0
  [✔] antixray-fabric-1.4.6+1.21.1
  [✔] appleskin-fabric-mc1.21-3.0.6
  [✔] archers-2.1.1+1.21.1
  [✔] attributefix-fabric-1.21.1-21.1.2
  [✔] azurelibarmor-fabric-1.21.1-2.3.13
  [✔] betterarcheology-1.2.0-1.21
  [✔] betterdays-1.21.1-3.1.1.1-FABRIC
  [✔] bookshelf-fabric-1.21.1-21.1.13
  [✔] c2me-fabric-mc1.21.1-0.3.0+alpha.0.295
  [✔] cardinal-components-api-6.1.1
  [✔] cloth-config-15.0.140-fabric
  [✔] comforts-fabric-9.0.3+1.21.1
  [✔] crashexploitfixer-fabric-1.1.0+1.21
  [✔] Creeperoverhaul-fabric-1.21-4.0.5
  [✔] Debugify-1.21.1+1.0
  [✘] dehydration-1.3.7.jar (disabled)
  [✔] emi-1.1.18+1.21.1+fabric
  [✔] entity_model_features_fabric_1.21-2.2.6
  [✔] entity_texture_features_fabric_1.21.1-6.2.7
  [✔] fabric-api-0.107.0+1.21.1
  [✔] fabric-language-kotlin-1.12.3+kotlin.2.0.21
  [✔] FarmersDelight-1.21.1-2.2.1+refabricated
  [✔] fasterrandom-5.1.0
  [✔] ferritecore-7.0.2-hotfix-fabric
  [✔] fleshz-1.6.1
  [✔] ForgeConfigAPIPort-v21.1.1-1.21.1-Fabric
  [✔] frostiful-1.3.3
  [✔] fusion-1.1.1-fabric-mc1.21
  [✔] fzzy_config-0.5.6+1.21
  [✔] geckolib-fabric-1.21.1-4.7
  [✔] GlitchCore-fabric-1.21.1-2.1.0.0
  [✔] handcrafted-fabric-1.21.1-4.0.2
  [✔] Jade-1.21.1-Fabric-15.8.3
  [✔] jewelry-2.0.2+1.21.1
  [✔] journeymap-fabric-1.21.1-6.0.0-beta.29
  [✔] Kiwi-1.21-Fabric-15.1.2
  [✔] krypton-0.2.8
  [✔] Ksyxis-1.3.2
  [✔] lithium-fabric-mc1.21.1-0.13.1
  [✔] lithostitched-fabric-1.21.1-1.3.4
  [✔] MapFrontiers-1.21-2.6.0p2-fabric
  [✔] modernfix-fabric-5.19.5+mc1.21.1
  [✔] moremobvariants-fabric+1.21-1.3.1.1
  [✔] mythicmetals-0.20.2+1.21
  [✔] NoChatReports-FABRIC-1.21.1-v2.9.1
  [✔] noisium-fabric-2.3.0+mc1.21-1.21.1
  [✔] owo-lib-0.12.15+1.21
  [✔] packetfixer-fabric-1.4.2-1.21-to-1.21.3
  [✔] paladins-2.1.1+1.21.1
  [✔] Palladium-1.21x-1.0.9.5
  [✔] player-animation-lib-fabric-2.0.0-alpha1+1.21
  [✔] polymorph-fabric-1.0.6+1.21.1
  [✔] potatoptimize-0.0.1-dev.7
  [✔] prickle-fabric-1.21.1-21.1.6
  [✔] PuzzlesLib-v21.1.22-1.21.1-Fabric
  [✔] randommobsizes-fabric-1.21.1-2.2.2
  [✔] ranged_weapon_api-2.0.4+1.21.1
  [✔] rechiseled-1.1.6b-fabric-mc1.21
  [✔] resourcefulconfig-fabric-1.21-3.0.8
  [✔] resourcefullib-fabric-1.21-3.0.11
  [✔] rogues-2.1.0+1.21.1
  [✔] runes-1.0.3+1.21.1
  [✔] ScalableLux-0.1.0+fabric.26c6e72-all
  [✔] scorchful-0.12.1
  [✔] sepals-1.0.4
  [✔] SereneSeasons-fabric-1.21.1-10.1.0.1
  [✔] servercore-fabric-1.5.5+1.21.1
  [✔] server-snapshot-performance-backports-1.21.1.2
  [✔] ShieldAPI-2.0.2+1.21.1
  [✔] SnowRealMagic-1.21.1-Fabric-11.0.12
  [✔] spark-1.10.109-fabric
  [✔] spell_engine-1.1.2+1.21.1
  [✔] spell_power-1.0.5+1.21.1
  [✔] structure_pool_api-1.1.3+1.21.1
  [✔] supermartijn642configlib-1.1.8-fabric-mc1.21
  [✔] supermartijn642corelib-1.1.17c-fabric-mc1.21
  [✔] tatimod-0.1.1-1.21.1
  [✔] tectonic-fabric-1.21.1-2.4.1a
  [✔] Terralith_1.21.x_v2.5.6
  [✔] thermoo-4.2.3
  [✔] thermoo-patches-3.3
  [✔] tide-fabric-1.21-1.3.4
  [✔] ToadLib-1.21x-1.1.6
  [✔] toms_storage_fabric-1.21-2.0.11
  [✔] travelersbackpack-fabric-1.21.1-10.1.3
  [✔] trinkets-3.10.0
  [✔] trowel-1.4.0
  [✔] vcinteraction-fabric-1.21.1-1.0.8
  [✔] villagerfix-1.0.4
  [✔] voicechat-fabric-1.21.1-2.5.25
  [✔] whiteout-0.13.0
  [✔] wizards-2.1.0+1.21.1
  [✔] YetAnotherConfigLib-3.6.1+1.21-fabric

### Logs

https://mclo.gs/XZaMJJc

Server console output after Client crashes:
```[Netty Server IO #9/ERROR]: Exception occurred in netty pipeline
java.net.SocketException: Connection reset
        at java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401) ~[?:?]
        at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434) ~[?:?]
        at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:254) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[minecraft-1.21.1.i0:0/:?]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[minecraft-1.21.1.i0:0/:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

### Minimal instance

- [ ] I have tested this on a minimal instance

### Performance and shader mods

- [X] I am using performance or shader mods

**Comment by shinycolororange** [comment]
same here even without other mods

**Comment by ChrysanthCow** [comment]
Could you try updating to the latest version of the mod? You are way out of date, and you are using our botched version of Porting Lib's item abilities code.

**Comment by qtGenn** [comment]
> Could you try updating to the latest version of the mod? You are way out of date

I apologize, I meant 2.2.1 and not 2.1.1 as evident in the mod list below. It was a typo.

**Comment by ChrysanthCow** [comment]
Alright, thanks for the clarification.

Ill have a fix out over the next few days, today is a tired day for me atm

**Comment by ChrysanthCow** [comment]
This issue persisted to Axes specifically, it was due to a faulty Porting Lib mixin, this has been fixed as of 2.2.2
